### ISSUE TITLE
It should be possible to update same version with different dependencies

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Warehouse#92]
Right now it is not possible to upload save package version of an app more than once. It should be possible to do so with different dependency of Starcounter.

For example the SignIn app version `3.2.0` should be compiled differently for `develop` and `pnext`. The app version is still `3.2.0`, but the dependency is different.

**Comment by andwah** [comment]
An example/question:

SignIn app version `3.2.0` with Compatiblility `~2.2.1.2998`
SignIn app version `3.2.0` with Compatiblility `~2.2.1.2999`

which version is the latest?

**Comment by miyconst** [comment]
@andwah it is not about which version is `latest`, but which version suits me.

If a person has Starcounter `2.2.1.2998` and App `3.2.0~2.2.1.2999` installed, then it should be possible to upgrade to `3.2.0~2.2.1.2998` because the installed version is not compatible with current Starcounter, and it is required to uninstall and install app again.

One click would be much easier. This is just sugaring, don't take it as a high priority.

**Comment by andwah** [comment]
I agree so which version suits me if i have starcounter version `2.2.1.2998`  (both version will work).

Today i select all versions and order by versiondate and pick the first compatible version. That will give me the latest version.

The compatible string just tells if the app is compatible with a starcounter version. We can not use that information to know which version is the latest.

I will think a bit more about this.. there is a simple solution for this...
